feature/dashboard/src/main/java/com/simprints/feature/dashboard/logout/usecase/LogoutUseCase.kt

```diff
@@ -14,13 +13,12 @@ internal class LogoutUseCase @Inject constructor(
     @ExternalScope private val externalScope: CoroutineScope,
 ) {
 
-    suspend operator fun invoke() {
-        authManager.signOut()
-
+    suspend operator fun invoke() = externalScope.launch {
         // Cancel all background sync
-        externalScope.launch {
```

Comment:
This will return the exact race condition that I fixed in the last PR. `authManager.signOut()` must be called in the caller's scope (view model in this case) to prevent navigation before the auth store is cleared out.

Reply:
Need to check if the external scope is required there, the cancellation might be fast enough to do synchronously as well.

Reply:
Maybe we need a comment about that to prevent other people from modifying this in the future.

Reply:
@luhmirin-s 
I updated this fun to Deferred 
now, the callers to the  logout can choose to wait until it finishes

@BurningAXE I added a comment

Reply:
Why wouldn't the caller want to wait for the operation to finish? IMO it runs the risk that somebody doesn't wait and introduces a bug this way.

Reply:
@BurningAXE Agreed. Waiting for the logout operation to complete is crucial in all scenarios. The lint warning serves as a safeguard, preventing callers from unintentionally ignoring the Deferred result.